app/src/main/java/com/example/trainingintive/dog_images_feature/data/repository/DogImageRepository.kt

```diff
@@ -24,4 +30,16 @@ class DogImageRepository @Inject constructor(
     fun update(imageUrl: DogImageUrl) = dogImageDao.update(imageUrl.toDatabaseEntity())
 
     fun remove(imageUrl: DogImageUrl) = dogImageDao.remove(imageUrl.toDatabaseEntity())
+
+    fun update(imageUrls: List<DogImageUrl>) = dogImageDao.update(imageUrls.map { it.toDatabaseEntity() })
+
+    private fun getLastPosition() = dogImageDao.count()
+
+    fun changeImagePosition(urls: List<DogImageUrl>, from: Int, to: Int): Completable {
+        val list = urls as MutableList
+        Collections.swap(list, from, to)
+        var i = 0
+        list.forEach { it.position = i++ }
```

Comment:
I don't get it :D Why do you perform this operation?

Reply:
it's updating item position in db. If I don't do that then when you reopen app items would be on the same positions like before changing